backend/compact-connect/lambdas/license-data/tests/unit/test_event_batch_writer.py

```diff
@@ -0,0 +1,144 @@
+import json
+from typing import List
+from unittest.mock import MagicMock
+from uuid import uuid4
+
+from tests import TstLambdas
+
+
+class TestEventBatchWriter(TstLambdas):
+    def test_write_big_batch(self):
+        from event_batch_writer import EventBatchWriter
+
+        put_count = []
+
+        def mock_put_items(Entries: List[dict]):  # pylint: disable=invalid-name
+            put_count.extend(Entries)
+            return {}
+
+        mock_client = MagicMock()
+        mock_client.put_events.side_effect = mock_put_items
+
+        with open('tests/resources/ingest/message.json', 'r') as f:
+            event = json.load(f)
+
+        with EventBatchWriter(client=mock_client) as writer:
+            # Send a bunch of messages, make sure each is sent
+            for _ in range(123):
+                writer.put_event(
+                    Entry=event
+                )
+
+        # Make sure each message was eventually sent
+        self.assertEqual(123, len(put_count))
+        # Make sure these were sent in three batches
+        self.assertEqual(3, mock_client.put_events.call_count)
+
+    def test_write_small_batch(self):
+        from event_batch_writer import EventBatchWriter
+
+        put_count = []
+
+        def mock_put_items(Entries: List[dict]):  # pylint: disable=invalid-name
+            put_count.extend(Entries)
+            return {}
+
+        mock_client = MagicMock()
+        mock_client.put_events.side_effect = mock_put_items
+
+        with open('tests/resources/ingest/message.json', 'r') as f:
+            event = json.load(f)
+
+        with EventBatchWriter(client=mock_client) as writer:
+            # Send a bunch of messages, make sure each is sent
+            for _ in range(23):
+                writer.put_event(
+                    Entry=event
+                )
```

Comment:
We could. I was erring on the side of repetative but explicit tests, rather than hiding the intended use of the unit we're testing via some form of abstraction.

Reply:
That's fair and reasonable. I wonder if we could de-duplicate some of the setup code- it just becomes tricky trying to identify what is the same and what is different. That said, I'm good punting this to another sprint.

Reply:
I'll see if I can find a better balance while I address your other comments. Getting lost in the the boiler plate is a sign that I've taken it too far, I think.